tests/test_transforms/test_encoders/test_mean_encoder_transform.py

```diff
@@ -114,12 +114,114 @@ def expected_ts_begin_nan_smooth_1() -> TSDataset:
 def expected_ts_begin_nan_smooth_2() -> TSDataset:
     df = generate_ar_df(start_time="2001-01-01", periods=6, n_segments=1)
     df.rename(columns={"target": "mean_encoded_regressor"}, inplace=True)
-    df["mean_encoded_regressor"] = [np.NaN, np.NaN, 2 / 3, 5 / 4, 5 / 3, 2.5]
+    df["mean_encoded_regressor"] = [np.NaN, np.NaN, np.NaN, 5 / 3, 5 / 3, 2.5]
 
     ts = TSDataset(df, freq="D")
     return ts
 
 
+@pytest.fixture
+def multiple_nan_target_new_category_ts() -> TSDataset:
+    """Fixture with several timestamp with NaN target for new category where there were no notna targets yet."""
+    df = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=5)
+    df["target"] = [np.NaN, 1.5, np.NaN, 3.0, 4.0]
+
+    df_exog = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=6)
+    df_exog.rename(columns={"target": "regressor"}, inplace=True)
+    df_exog["regressor"] = ["A", "B", "A", "A", "B", "C"]
+
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future="all")
+
+    return ts
+
+
+@pytest.fixture
+def expected_multiple_nan_target_new_category_ts() -> TSDataset:
+    df = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=5)
+    df.rename(columns={"target": "regressor_mean"}, inplace=True)
+    df["regressor_mean"] = [np.NaN, np.NaN, np.NaN, np.NaN, 1.5]
+
+    ts = TSDataset(df=df, freq="D")
+
+    return ts
+
+
+@pytest.fixture
+def multiple_nan_target_old_category_ts() -> TSDataset:
+    """Fixture with several timestamp with NaN target for category where there was already a notna target."""
```

Comment:
I think it is better to write:
```
Fixture with segment having multiple NaN targets after first non-NaN target value.

Look at the segment ``B``.
```

Reply:
Do we really need segment `A` here?

Reply:
My motivation was to check each case when there is more that one unique category.
Okey, let't combine these tests.